### PR TITLE
pr-tool: update LICENSE.txt

### PR DESCRIPTION
**Pull request by joergsteffens**
### Thank you for contributing to the Bareos Project!

 Add update-license subcommand to the pr-tool.
 It will be called automatically when merging a PR.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- ~Required backport PRs have been created~

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- ~Required documentation changes are present and part of the PR~

### CODE REVIEW

**Review by arogge — CHANGES_REQUESTED** [review]
I found some minor issues with the copyright headers.
Besides that there is a small performance gain possible by omitting an API request and pylint brought up a few remarks.

File: devtools/pip-tools/license_utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/license_utils/__init__.py:19]: added.

File: devtools/pip-tools/license_utils/update_license.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/license_utils/update_license.py:21]: done.

File: devtools/pip-tools/license_utils/update_license.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    try:
+        generate_license_file(template_filename, target_filename)
+    except FileNotFoundError:
+        print(f"Could not find file {template_filename}", file=stderr)
+        return 2
+    return 0
+
+
+if __name__ == "__main__":
+    exit(main())
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/license_utils/update_license.py]: done.

File: devtools/pip-tools/pr_tool/main.py [hunk 4]
```diff
@@ -743,6 +775,11 @@ def main():
             ret = False
 
         handle_ret(ret)
+    elif args.subcommand == "update-license":
+        if update_license_file(repo) in [None, True]:
+            exit(0)
+        else:
+            exit(1)
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/pr_tool/main.py]: Good point. Moved it to an earlier stage.

File: devtools/pip-tools/license_utils/__init__.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 202-2024 Bareos GmbH & Co. KG
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/license_utils/__init__.py]: done.

File: devtools/pip-tools/setup.py [hunk 6]
```diff
@@ -34,6 +34,7 @@
             "pr-tool=pr_tool.main:main",
             "update-changelog-links=changelog_utils.update_links:main",
             "add-changelog-entry=changelog_utils.add_entry:main",
+            "update-license=license_utils.update_license:main",
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply devtools/pip-tools/setup.py:37]: done.

**Comment by joergsteffens** [comment]
I just discovered. that github now has 2 License entries. I assume, one for LICENSE.txt and on for LICENSE.template (https://github.com/bareos/bareos?tab=License-2-ov-file).
I  assume, it is best, if I move the tempate file to another directory or rename the file.

**Pull request merged**